crypto/rand/rand_vxworks.c

```diff
@@ -0,0 +1,164 @@
+ */
+ 
+#ifndef _GNU_SOURCE
+# define _GNU_SOURCE
+#endif
+#include <openssl/opensslconf.h>
+
+#ifndef OPENSSL_SYS_VXWORKS
+NON_EMPTY_TRANSLATION_UNIT
+#else
+# include <openssl/rand.h>
+# include "rand_lcl.h"
+# include "internal/rand_int.h"
+# include "internal/cryptlib.h"
+# include <version.h>
+# include <taskLib.h>
+
+# if defined(OPENSSL_RAND_SEED_NONE)
+/* none means none */
+#  undef OPENSSL_RAND_SEED_OS
+# endif
+
+# if defined(OPENSSL_RAND_SEED_OS) 
+#  if _WRS_VXWORKS_MAJOR >= 7
+#    define RAND_SEED_VXRANDLIB
+#  else
+#    error "VxWorks <7 only support RAND_SEED_NONE"
+#  endif
+# endif
+
+# if defined(RAND_SEED_VXRANDLIB)    
+#  include <randomNumGen.h>
+# endif
+
+/* Macro to convert two thirty two bit values into a sixty four bit one */
+# define TWO32TO64(a, b) ((((uint64_t)(a)) << 32) + (b))
+
+static uint64_t get_time_stamp(void)
+{
+    struct timespec ts;
+
+    if (clock_gettime(CLOCK_REALTIME, &ts) == 0)
+        return TWO32TO64(ts.tv_sec, ts.tv_nsec);
+    return time(NULL);
+}
+
+static uint64_t get_timer_bits(void)
+{
+    uint64_t res = OPENSSL_rdtsc();
+
+    if (res != 0)
+        return res;
+
+    {
+        struct timespec ts;
+
+        if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0)
+            return TWO32TO64(ts.tv_sec, ts.tv_nsec);
+    }
+    return time(NULL);
+}
+
+/* 
+ *	empty implementation 
+ *  vxworks does not need to init/cleanup or keep open the random lib
+ */
+int rand_pool_init(void)
+{
+    return 1;
+}
+
+void rand_pool_cleanup(void)
+{
+}
+
+void rand_pool_keep_random_devices_open(int keep)
+{
+}
+
+int rand_pool_add_additional_data(RAND_POOL *pool)
+{
+    struct {
+        CRYPTO_THREAD_ID tid;
+        uint64_t time;
+    } data = { 0 };
+
+    /*
+     * Add some noise from the thread id and a high resolution timer.
+     * The thread id adds a little randomness if the drbg is accessed
+     * concurrently (which is the case for the <master> drbg).
+     */
+    data.tid = CRYPTO_THREAD_get_current_id();
+    data.time = get_timer_bits();
+
+    return rand_pool_add(pool, (unsigned char *)&data, sizeof(data), 0);
+}
+
+int rand_pool_add_nonce_data(RAND_POOL *pool)
+{
+    struct {
+        pid_t pid;
+        CRYPTO_THREAD_ID tid;
+        uint64_t time;
+    } data = { 0 };
+
+    /*
+     * Add process id, thread id, and a high resolution timestamp to
+     * ensure that the nonce is unique whith high probability for
+     * different process instances.
+     */
+    data.pid = getpid();
+    data.tid = CRYPTO_THREAD_get_current_id();
+
+    data.time = get_time_stamp();
+
+    return rand_pool_add(pool, (unsigned char *)&data, sizeof(data), 0);
+}
+
+size_t rand_pool_acquire_entropy(RAND_POOL *pool)
+{
+# if defined(RAND_SEED_VXRANDLIB)
+/* vxRandLib based entropy method */
+    size_t bytes_needed;
+
+    bytes_needed = rand_pool_bytes_needed(pool, 1 /*entropy_factor*/);
+    if (bytes_needed > 0) 
+    {
+        int retryCount = 0;
+        STATUS result = ERROR;
+        

```

Comment:
the empty line needs to be after "unsinged char buffer;"  the idea is to spearate the variable declaration
section from the code section, we still adhere to C90.